Classes/Event/EventPublisher.php

```diff
@@ -45,46 +47,65 @@ class EventPublisher
 
     /**
      * @param EventStore $eventStore
-     * @param EventBus $eventBus
+     * @param EventListenerLocator $eventListenerLocator
      * @param PropertyMapper $propertyMapper
      * @param EventTypeResolver $eventTypeResolver
      */
-    public function __construct(EventStore $eventStore, EventBus $eventBus, PropertyMapper $propertyMapper, EventTypeResolver $eventTypeResolver)
+    public function __construct(EventStore $eventStore, EventListenerLocator $eventListenerLocator, PropertyMapper $propertyMapper, EventTypeResolver $eventTypeResolver)
     {
         $this->eventStore = $eventStore;
-        $this->eventBus = $eventBus;
+        $this->eventListenerLocator = $eventListenerLocator;
         $this->propertyMapper = $propertyMapper;
         $this->eventTypeResolver = $eventTypeResolver;
     }
 
 
     /**
      * @param string $streamName
-     * @param EventWithMetadata $eventWithMetadata
+     * @param EventInterface $event
      * @param int $expectedVersion
      */
-    public function publish(string $streamName, EventWithMetadata $eventWithMetadata, int $expectedVersion = ExpectedVersion::ANY)
+    public function publish(string $streamName, EventInterface $event, int $expectedVersion = ExpectedVersion::ANY)
     {
-        $this->publishMany($streamName, [$eventWithMetadata], $expectedVersion);
+        $this->publishMany($streamName, [$event], $expectedVersion);
     }
 
     /**
      * @param string $streamName
-     * @param EventWithMetadata[] $eventsWithMetadata
+     * @param EventInterface[] $events
      * @param int $expectedVersion
      */
-    public function publishMany(string $streamName, array $eventsWithMetadata, int $expectedVersion = ExpectedVersion::ANY)
+    public function publishMany(string $streamName, array $events, int $expectedVersion = ExpectedVersion::ANY)
```

Comment:
I previously thought that we need to be able to pass in metadata here for migration: But in order to *record* events after the fact one shouldn't use the `EventPublisher` to begin with but interact with the `EventStore` directly